python/cog.py

```diff
@@ -117,11 +117,58 @@ async def slash_search(self, ctx: SlashContext, **kwargs):
         if isinstance(files, str):
             await ctx.send(content=files, hidden=True)
             return
-        author = ctx
+
+        recipient = ctx
         if kwargs.get("dm"):
-            author = ctx.author
+            recipient = ctx.author
             await ctx.send("DM'ing your files...", hidden=True)
-        await self.send_files_as_message(author, files)
+
+        return (recipient, files)
+
+    @cog_ext.cog_slash(
+        name="search",
+        description="Search for files.",
+        options=search_options,
+        guild_ids=guild_ids if getattr(CONFIG, "DB_NAME", "production") == "testing" else []
+    )
+    async def slash_search(self, ctx: SlashContext, **kwargs):
+        """
+        Responds to `/search`. Tries to display docs related to a query from ElasticSearch.
+
+        Args:
+            ctx: The SlashContext from which the command originated
+            filename: A str of the filename to query for.
+            DM: A bool for whether to dm the author the results.
+        """
+        recipient, files = await self.locate(ctx, **kwargs)
+        await self.send_files_as_message(recipient, files)
+
+    @cog_ext.cog_slash(
+        name="export",
+        description="Get a Python export script to download the files returned in a search to your computer.",
+        options=search_options,
+        guild_ids=guild_ids if getattr(CONFIG, "DB_NAME", "production") == "testing" else []
+    )
+    async def slash_export(self, ctx: SlashContext, **kwargs):
+        """
+        Responds to `/export`. Tries get a download script for all files related to a query from ElasticSearch.
+
+        Args:
+            ctx: The SlashContext from which the command originated.
+            filename: A str of the filename to query for, if any.
+            DM: A bool for whether to dm the author the results.
+        """
+        recipient, files = await self.locate(ctx, **kwargs)
+        channels = {}
+        for c in ctx.guild.channels:
```

Comment:
Currently this will expose all channels on the server, regardless of whether the user has the necessary permissions to even know the channels exists. If this is a problem, we can check which channels are actually necessary to include by looking at the files list.